src/ripple/nodestore/RetryFinalize.h

```diff
@@ -0,0 +1,61 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2020 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#ifndef RIPPLE_NODESTORE_RETRYFINALIZE_H_INCLUDED
+#define RIPPLE_NODESTORE_RETRYFINALIZE_H_INCLUDED
+
+#include <ripple/app/main/Application.h>
+#include <functional>
+
+namespace ripple {
+namespace NodeStore {
+
+class RetryFinalize
+{
+public:
+    using retryFunction = std::function<void(std::uint32_t shardIndex)>;
+
+    RetryFinalize() = default;
+
+    bool
+    retry(Application& app, retryFunction f, std::uint32_t shardIndex);
+
+    // Must match the imported shard's last ledger hash
+    uint256 referenceHash{0};
+
+private:
+    using waitable_timer =
+        boost::asio::basic_waitable_timer<std::chrono::steady_clock>;
+
+    static constexpr std::chrono::seconds retryInterval_ =
+        std::chrono::seconds{60};
+
+    // Maximum attempts to retrieve a shard's last ledger hash
+    static constexpr uint32_t maxAttempts_{5};
```

Comment:
May be use std::uint32_t for uniformity?

Reply:
Thanks for the comment. I already fixed this when Scott pointed out the `namespace` issue. This class has been renamed to `ShardVerificationScheduler` and moved to "ripple/rpc/" so you can check there.